Framework/Core/src/CommonDataProcessors.cxx

```diff
@@ -63,6 +68,225 @@ struct InputObject {
 const static std::unordered_map<OutputObjHandlingPolicy, std::string> ROOTfileNames = {{OutputObjHandlingPolicy::AnalysisObject, "AnalysisResults.root"},
                                                                                        {OutputObjHandlingPolicy::QAObject, "QAResults.root"}};
 
+
+void CommonDataProcessors::DataRef2Tree(TFile* f, InputRecord inprec, DataRef ref)
+{
+
+  // check healt of DataRef
+  if (!ref.header) {
+    LOG(ERROR) << "Header not found";
+    return;
+  }
+
+  if (!ref.payload) {
+    LOG(ERROR) << "Payload not found";
+    return;
+  }
+
+  auto datah = o2::header::get<o2::header::DataHeader*>(ref.header);
+  if (!datah) {
+    LOG(ERROR) << "No data header in stack";
+    return;
+  }
+
+  // check if serialisation method is arrow
+  if (datah->payloadSerializationMethod == o2::header::gSerializationMethodArrow)
+  {
+    
+    // get the table name
+    auto treename = datah->dataDescription.as<std::string>();
+    LOG(DEBUG) << "The tree name is " << treename << std::endl; 
+    
+    // get the TableConsumer and convert it into an arrow table
+    auto s = inprec.get<TableConsumer>(ref.spec->binding);
+    auto table = s->asArrowTable();
+    LOG(DEBUG) << "Number of columns " << table->num_columns() << std::endl; 
+    LOG(DEBUG) << "Number of rows     " << table->num_rows() << std::endl;
+    if (table->num_columns()==0 || table->num_rows()==0)
+      return;
+
+
+    // create the corresponding tree
+    TTree *T = new TTree(treename.c_str(),treename.c_str());
+    
+    // loop over the columns
+    for (int ii=0; ii<table->num_columns(); ii++)
+    { 
+    
+      // get column information
+      auto col = table->column(ii);
+      auto chs = col->data()->chunks();
+      LOG(DEBUG) << "number of chunks " << chs.size() << std::endl;
+      LOG(DEBUG)  << " column type " << col->type()->ToString() << std::endl;
+    
+      // what follows is an ugly switch
+      // the cases cover the different data types of the branches
+      auto cdt = col->type()->id();
+      switch (cdt) {
+      case arrow::Type::type::FLOAT:
+        {
+          LOG(DEBUG) << "case FLOAT" << std::endl;
+          Float_t *dbuf = new Float_t();
```

Comment:
Why don't you simply use a Float_t and take the address of it? That will remove the need for a `new / delete` pair.